erdos/ros/ros_input_data_stream.py

```diff
@@ -61,13 +64,20 @@ def _on_msg(self, msg):
             # Also, maintain the lowest watermark observed.
             low_watermark = msg.timestamp
```

Comment:
Does this work when we're recovering? I'm thinking of the case when we rollback all operators to time t, but ROS still stores in its queues watermarks with a higher time (e.g., t + 2). These watermarks will be delivered after operator recovery, and thus we may end up marking time t + 2 as complete before we should have (e.g., after we received the messages tagged with t + 1 and  t + 2 from the recovered operators).